applications/StructuralMechanicsApplication/custom_constitutive/plane_stress_d_plus_d_minus_damage_masonry_2d.cpp

```diff
@@ -0,0 +1,688 @@
+	DamageDPlusDMinusMasonry2DLaw::CalculateExponentialDamageTension(UniaxialStress, rThreshold, damage_parameter, CharacteristicLength, rValues, rDamage);
+	
+	rPredictiveStressVector *= (1.0 - rDamage);
+}
+/***********************************************************************************/
+/***********************************************************************************/		
+void DamageDPlusDMinusMasonry2DLaw::CalculateDamageParameterTension(
+	ConstitutiveLaw::Parameters& rValues,
+    double& rAParameter,
+    const double CharacteristicLength)
+{
+	const Properties& r_material_properties = rValues.GetMaterialProperties();
+	
+	const double Gf = r_material_properties[FRACTURE_ENERGY_TENSION];
+	const double E = r_material_properties[YOUNG_MODULUS];
+	const double yield_tension = = r_material_properties[YIELD_STRESS_TENSION];
+	const double l_mat = 2.0 * E * Gf / (std::pow(yield_tension, 2));
+	rAParameter = 2.0 * (CharacteristicLength / (l_mat - CharacteristicLength);
+	KRATOS_ERROR_IF(rAParameter < 0.0) << "FRACTURE_ENERGY_TENSION is too low, increase it ..." << std::endl;	
+}
+/***********************************************************************************/
+/***********************************************************************************/
+void DamageDPlusDMinusMasonry2DLaw::CalculateExponentialDamageTension(
+	const double UniaxialStress,
+	const double Threshold,
+	const double DamageParameter,
+	const double CharacteristicLength,
+	ConstitutiveLaw::Parameters& rValues,
+	double& rDamage)
+{
+	const Properties& r_material_prop = rValues.GetMaterialProperties();
+	const double initial_threshold = r_material_properties["YIELD_STRESS_TENSION"]
+	rDamage = 1.0 - (initial_threshold / UniaxialStress) * std::exp(DamageParameter * (1.0 - (UniaxialStress / initial_threshold)))
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+void DamageDPlusDMinusMasonry2DLaw::IntegrateStressVectorCompression(
+	array_1d<double,3>& rPredictiveStressVector,
+	const double UniaxialStress,
+    double& rDamage,
+    double& rThreshold,
+    ConstitutiveLaw::Parameters& rValues,
+    const double CharacteristicLength)
+{
+	DamageDPlusDMinusMasonry2DLaw::CalculateBezier3DamageCompression(UniaxialStress, rDamage, rThreshold, CharacteristicLength, rValues);
+	
+	rPredictiveStressVector *= (1.0 - rDamage);
+}
+/***********************************************************************************/
+/***********************************************************************************/
+void DamageDPlusDMinusMasonry2DLaw::CalculateBezier3DamageCompression(
+	const double UniaxialStress,
+	double& rDamage,
+	double& rThreshold,
+	const double CharacteristicLength,
+	ConstitutiveLaw::Parameters& rValues)
+{
+	// Call the Material Properties 
+	const Properties& r_material_prop = rValues.GetMaterialProperties();

```

Comment:
Here you call the rValues.GetMaterialProperties() but then you use tha variable r_materail_properties instead of r_material_prop. This compiles?